xknx/management/management.py

```diff
@@ -65,6 +73,10 @@ def process(self, telegram: Telegram) -> None:
                 self.xknx.cemi_handler.send_telegram(disconnect)
             )
             return
+        if isinstance(telegram.tpci, TDataBroadcast):
+            for callback in self._rx_broadcast_cb:
+                callback(telegram)
+            return
```

Comment:
Would we need multiple parallel instances of broadcast listeners or would a singleton be enough?

I'd think it would be cleaner if the gathering and forwarding of telegrams was done in this class, rather than externally with callbacks. Like with an async-generator. The procedures module would be easier to read then and we woudn't have to think about cleaning up callbacks.
Also we wouldn't need any `asyncio.sleep` in the procedure, but could leverage `asyncio_timeout` in the generator.

Reply:
I have drafted an example. Didn't test it yet, but I think it should work fine.
see https://github.com/XKNX/xknx/commit/af8290c5d2f4e45c3e729e835a24c0b1c8d8bfe3
It would even support multiple parallel Broadcast contexts and can be easily extended to check APCI type to only yield specific telegrams.
Let me know what you think.

Reply:
Great! Thank you! I got it working with a few minor modifications.